java/src/jmri/NamedBeanPropertyDescriptor.java

```diff
@@ -0,0 +1,51 @@
+package jmri;
+
+/**
+ * Describes metadata about a given property key for a NamedBean. This metadata is used by the
+ * BeanTable actions to display and edit the properties in question.
+ *
+ * Created by bracz on 1/19/18.
+ */
+
+public abstract class NamedBeanPropertyDescriptor {
```

Comment:
Could you be more specific about what do you want as the type argument? Today nobody casts this object, so I'm not sure which casting you want to avoid.
The TableDataModel objects are setting and getting their properties using Object pointers, because a table is heterogenous, i.e., different columns have different types.
Also a bunch of NmedBeanPropertyDesriptors will be returned in a list, each possibly of a differently typed column.

Reply:
> The TableDataModel objects are setting and getting their properties using Object pointers, because a table is heterogenous, i.e., different columns have different types.
Also a bunch of NmedBeanPropertyDesriptors will be returned in a list, each possibly of a differently typed column.

By making the NamedBeanPropertyDescriptor use Generics, you avoid the need to create special subclasses for every single supported type of property, whereas now you are building ourself a situation where that can't be done.

The assumption that properties of NamedBeans are only used in tables is demonstrably invalid; therefor support for describing the property in a UI needs to support use cases outside a table.

Reply:
I made a change to use generics in the NamedBeanPropertyDescriptor. Let me know if this is what you thought.

Reply:
That is what I thought, but also, it appears that you are potentially replacing non-String properties in NamedBeans with Strings with renderProperty in [line 373 of BeanTableDataModel](https://github.com/JMRI/JMRI/pull/4761/files#diff-eaab12bf6d85fe1c5d53ab81f7018fa0R373) which suggests that `renderProperty` should return `E` instead of `String`.

Reply:
The render/parse method pairs convert to and from strings by design. We are saving these strings in the XML so they are persistent.
We can extend the capabilities of the XML storage towards generic properties, but for the moment the PR is slowballing and I'd like to keep the scope limited to existing feature set.

Reply:
NamedBeans *do not* retain XML persistence information, and the properties under NamedBeans must match their expected types. XML handling for NamedBeans is handled by the managers for the various NamedBean types under `jmri.jmrix`, so nothing in the table handling and GUI rendering code should be concerned with that.

This code breaks that entirely, and must be fixed before it lands, or else you will have written Olcb-specific code into the NamedBean tables, which we rather explicitly do not want.